haystack/preview/components/generators/openai.py

```diff
@@ -0,0 +1,284 @@
+import dataclasses
+import logging
+import os
+from typing import Optional, List, Callable, Dict, Any
+
+import openai
+from openai.openai_object import OpenAIObject
+
+from haystack.preview import component, default_from_dict, default_to_dict
+from haystack.preview.components.generators.utils import serialize_callback_handler, deserialize_callback_handler
+from haystack.preview.dataclasses import StreamingChunk, ChatMessage
+
+logger = logging.getLogger(__name__)
+
+
+API_BASE_URL = "https://api.openai.com/v1"
+
+
+@component
+class GPTGenerator:
+    """
+    Enables text generation using OpenAI's large language models (LLMs). It supports gpt-4 and gpt-3.5-turbo
+    family of models.
+
+    Users can pass any text generation parameters valid for the `openai.ChatCompletion.create` method
+    directly to this component via the `**generation_kwargs` parameter in __init__ or the `**generation_kwargs`
+    parameter in `run` method.
+
+    For more details on the parameters supported by the OpenAI API, refer to the OpenAI
+    [documentation](https://platform.openai.com/docs/api-reference/chat).
+
+    ```python
+    from haystack.preview.components.generators import GPTGenerator
+    client = GPTChatGenerator()
```

Comment:
client = GPTChatGenerator()
Comment
```suggestion
    client = GPTGenerator()
```